tools/kleaver/main.cpp

```diff
@@ -95,6 +95,11 @@ namespace {
   llvm::cl::opt<std::string> directoryToWriteQueryLogs("query-log-dir",llvm::cl::desc("The folder to write query logs to. Defaults is current working directory."),
 		                                               llvm::cl::init("."));
 
+  llvm::cl::opt<bool>
+  QueryIndependence("assume-query-independence",
```

Comment:
I really don't like this name. It sounds like it has something to do with the ``IndependenceSolver`` even though it doesn't.

The option probably ought to be called something like ``-persistent-array-symbol-table`` and would be on by default.